Sources/WebSocketKit/WebSocket.swift

```diff
@@ -63,6 +75,32 @@ public final class WebSocket {
         self.onPingCallback = callback
     }
 
+    public func onEvent(_ identifier: String, _ handler: @escaping (WebSocket) -> Void) {
+        events[identifier] = { ws, data in
+            handler(ws)
+        }
+    }
+
+    public func onEvent<T>(_ identifier: String, _ type: T.Type, _ handler: @escaping (WebSocket, T) -> Void) where T: Codable {
+        onEvent(identifier, handler)
+    }
+
+    public func onEvent<T>(_ identifier: String, _ handler: @escaping (WebSocket, T) -> Void) where T: Codable {
+        events[identifier] = { [weak self] ws, data in
+            guard let self = self else { return }
+            do {
+                let res = try JSONDecoder().decode(WebSocketEvent<T>.self, from: data)
+                if let data = res.data {
+                    handler(ws, data)
+                } else {
+                    self.logger.warning("Unable to unwrap data for event `\(identifier)`, because it is unexpectedly nil. Please use another `bind` method which support optional payload to avoid this message.")
```

Comment:
This should be trace most likely. Warnings should rarely be thrown by libraries. See the [Logging Guide](https://github.com/swift-server/guides/blob/main/docs/libs/log-levels.md) for more details